src/set_theory/cardinal/ordinal.lean

```diff
@@ -696,6 +696,24 @@ eq.symm $ le_antisymm (mk_le_of_injective (λ x y, finset.singleton_inj.1)) $
 calc #(finset α) ≤ #(list α) : mk_le_of_surjective list.to_finset_surjective
 ... = #α : mk_list_eq_mk α
 
+@[simp] lemma mk_finsupp_of_infinite (α : Type u) (β : Type v) [infinite α] [has_zero β]
+  [nontrivial β] : #(α →₀ β) = max (lift.{v} (#α)) (lift.{u} (#β)) :=
+begin
+  apply le_antisymm,
+  { calc #(α →₀ β) ≤ # (finset (α × β)) : mk_le_of_injective (finsupp.graph_injective α β)
+    ... = #(α × β) : mk_finset_of_infinite _
+    ... = max (lift.{v} (#α)) (lift.{u} (#β)) :
+      by rw [mk_prod, mul_eq_max_of_aleph_0_le_left]; simp },
+  { apply max_le,
+    { cases exists_ne (0 : β) with b hb,
+      rw [←lift_id (# (α →₀ β)), ←lift_umax.{u v}],
+      apply embedding.cardinal_lift_mk_le.{u (max u v) v},
+      exact ⟨_, finsupp.single_left_injective hb⟩ },
+    { rw [←lift_id (# (α →₀ β)), ←lift_umax.{v u}],
+      apply embedding.cardinal_lift_mk_le.{v (max u v) u},
+      exact ⟨_, finsupp.single_injective (classical.arbitrary α)⟩ } }
```

Comment:
then you can golf here:
```suggestion
      exact lift_mk_le_mk.2 ⟨⟨_, finsupp.single_left_injective hb⟩⟩ },
    { inhabit α,
      exact lift_mk_le_mk.2 ⟨⟨_, finsupp.single_injective default⟩⟩ } }
```

Reply:
I golfed the theorem by using your approach, but without the new theorem you mentioned.